From f8cdfba26c670281449348781b88117a24068fbe Mon Sep 17 00:00:00 2001
From: md_5 <md_5@live.com.au>
Date: Fri, 28 Jun 2013 19:52:54 +1000
Subject: [PATCH] Disallow Interaction With Self


diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 6e7b8792a..de3f02c25 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -2024,6 +2024,13 @@ public class PlayerConnection implements PacketListenerPlayIn {
         if (this.player.isFrozen()) return; // CraftBukkit
         WorldServer worldserver = this.player.getWorldServer();
         Entity entity = packetplayinuseentity.a((World) worldserver);
+        // Spigot Start
+        if ( entity == player && !player.isSpectator() )
+        {
+            disconnect( "Cannot interact with self!" );
+            return;
+        }
+        // Spigot End
 
         this.player.resetIdleTimer();
         this.player.setSneaking(packetplayinuseentity.e());
-- 
2.24.1.windows.2

