From e37bcfcc7c730b0c7c3d14cfacd80c6146f21206 Mon Sep 17 00:00:00 2001
From: md_5 <git@md-5.net>
Date: Wed, 22 Jul 2015 19:04:37 +1000
Subject: [PATCH] Clear Packet Queue on Disconnect


diff --git a/src/main/java/net/minecraft/network/NetworkManager.java b/src/main/java/net/minecraft/network/NetworkManager.java
index 2d73e9f60..82d8a163d 100644
--- a/src/main/java/net/minecraft/network/NetworkManager.java
+++ b/src/main/java/net/minecraft/network/NetworkManager.java
@@ -332,6 +332,7 @@ public class NetworkManager extends SimpleChannelInboundHandler<Packet<?>> {
                 } else if (this.j() != null) {
                     this.j().a(new ChatMessage("multiplayer.disconnect.generic"));
                 }
+                this.packetQueue.clear(); // Free up packet queue.
             }
 
         }
diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 9166b4184..861f7d3f3 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -1671,7 +1671,7 @@ public class PlayerConnection implements PacketListenerPlayIn {
         }
 
         // CraftBukkit start
-        if (packet == null) {
+        if (packet == null || this.processedDisconnect) { // Spigot
             return;
         } else if (packet instanceof PacketPlayOutSpawnPosition) {
             PacketPlayOutSpawnPosition packet6 = (PacketPlayOutSpawnPosition) packet;
-- 
2.24.1.windows.2

